cli/generators/crud/templates/components/addPage.js

```diff
@@ -0,0 +1,72 @@
+import React, { PropTypes } from 'react';
+import { browserHistory } from 'react-router';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import toastr from 'toastr';
+import * as <%= name %>Actions from '../../actions/<%= name %>Actions';
+import <%= ucName %>Form from './<%= ucName %>Form';
+
+class <%= ucName %>AddPage extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+
+    this.state = {
+      <%= name %>: Object.assign({}, props.<%= name %>),
+      saving: false
+    };
+
+    this.handleSave = this.handleSave.bind(this);
+  }
+
+  handleSave(<%= name %>) {
+    this.setState({ saving: true });
+    this.props.actions.create<%= ucName %>(<%= name %>)
+      .then(() => {
+        this.setState({ saving: false });
+        toastr.success('<%= ucName %> created successfully');
+        browserHistory.push('/app/<%= pluralizedName %>');
+      })
+      .catch(error => {
+        this.setState({ saving: false });
```

Comment:
If you add another `.then` after the catch, does it get called on rejection? Maybe you can refactor the `this.setState({ saving: false });` to a single place, if you do that.

(Sorry I'm asking these basic questions. Several frameworks used to handle this in different ways and now I'm all mixed up.)

This comment also applies for all the places where this is done later on.